offchain-modules/lib/src/util/eth_util.rs

```diff
@@ -34,76 +25,69 @@ impl Web3Client {
     pub fn client(&mut self) -> &mut Web3<Http> {
         &mut self.client
     }
-    pub fn chain_id(&mut self) -> &u32 {
-        &self.chain_id
-    }
 
-    pub fn send_transaction(
+    pub async fn send_transaction(
         &mut self,
         from: H160,
         to: H160,
         key_path: String,
         data: Vec<u8>,
-    ) -> H256 {
-        let nonce = self
-            .client()
-            .eth()
-            .transaction_count(from, None)
-            .wait()
-            .unwrap();
-        let tx = make_transaction(to, nonce, data);
+        eth_value: U256,
+    ) -> Result<H256> {
+        let nonce = self.client().eth().transaction_count(from, None).await?;
+        dbg!(&nonce);
+        let chain_id = self.client().eth().chain_id().await?;
+        dbg!(&chain_id);
+        let tx = make_transaction(to, nonce, data, eth_value);
         let signed_tx = tx.sign(
             &parse_private_key(key_path.as_str()).unwrap(),
-            &self.chain_id,
+            &chain_id.as_u32(),
         );
         let tx_hash = self
             .client()
             .eth()
+            // .send_raw_transaction_with_confirmation(Bytes::from(signed_tx), Duration::new(5, 100), 10)
             .send_raw_transaction(Bytes::from(signed_tx))
-            .wait()
-            .unwrap();
-        println!("tx hash: {:?}", tx_hash);
-        tx_hash
+            .await?;
+        log::debug!("tx hash: {:?}", tx_hash);
+        Ok(tx_hash)
     }
 
-    pub fn get_blocks(&mut self, start: usize, stop: usize) -> (Vec<Vec<u8>>, Vec<H256>) {
-        let futures = (start..stop)
-            .map(|i| self.client.eth().block((i as u64).into()))
-            .collect::<Vec<_>>();
-        let block_headers = join_all(futures).wait().unwrap();
-
-        let mut blocks: Vec<Vec<u8>> = vec![];
-        let mut hashes: Vec<H256> = vec![];
-        for block_header in block_headers {
-            let mut stream = RlpStream::new();
-            rlp_append(&block_header.clone().unwrap(), &mut stream);
-            blocks.push(stream.out());
-            hashes.push(H256(block_header.clone().unwrap().hash.unwrap().0));
-        }
-        for item in &blocks {
-            println!("header rlp: {:?}", hex::encode(item.clone()));
-        }
-
-        (blocks, hashes)
+    pub async fn get_header_rlp_with_hash(&mut self, hash: H256) -> String {
+        let block_header = self
+            .client
+            .eth()
+            .block(BlockId::Hash(hash))
+            .await
+            .expect("invalid header");
+        let mut stream = RlpStream::new();
+        rlp_append(&block_header.unwrap(), &mut stream);
+        let header_vec = stream.out();
+        hex::encode(header_vec)
```

Comment:
If something goes wrong with the network or header_data, the function will panic. I think we should handle the error rather than panic.